### DISCUSSION TITLE
Testers needed

### DISCUSSION THREAD

**Discussion by rrirower** [rrirower/Channel-Feed-Manager#13] [General]
If you are interested in testing a new software product to create and manage your channel feed, please add to this discussion with your email address.  A link to an installer program will be sent when it is available.

**Comment by DaveBarnitz** [comment]
Interested

**Reply by rrirower** [reply]
Hi Dave.. Please provide an email address.

**Reply by DaveBarnitz** [reply]
sry, name then @gmail.com  Thanks

**Reply by DaveBarnitz** [reply]
firstLast @gmail.com fwiw

**Reply by rrirower** [reply]
Thanks.. Got it

**Reply by DaveBarnitz** [reply]
Two preliminary issues or recommendations:

1. I found it not too intuitive to 'right click' to add Videos & Ad Breaks etc. Nothing said to do so and I was double clicking, filling out the remaining fields hoping it would allow me access. Maybe writing next to 'Videos in this content' place '(right click to add)'

2. I tried exporting a sample JSON and it said it successfully exported to the "Channels/Export folder" but I don't know where that is?   Where do I go to go see it?

**Reply by DaveBarnitz** [reply]
Small issue, but once in the Video field I pasted the url, however the placeholder 'https://' stayed in the field and was at the tail end of my URL. Had I not noticed it likely would've returned an invalid url from Roku. Can this just be an empty string of text, instead of actually staying there (like when credit card processors read MM/DD etc)

Also, I feel the 'Tags' are a bit locked in to the Categories, with little leeway. For example, sometimes I create Categories for 'Popular' or 'Christmas' and move videos in & out. Kinda choosing which ones go into 'Popular' whenever I feel like it. Also maybe use that tag for 2 or 3 separate Categories. Any reason Tags isn't its own field where we data entry them per video?

**Reply by rrirower** [reply]
Keep the feedback coming.  I'll take a look at your concerns.

**Reply by DaveBarnitz** [reply]
Running into 2 issues in creating a Series.

1,.  In the Series tab, under Credits, I right click and 'Add New Item'. The default text gets added [Name of person] but I seem unable to edit this?  (tried double clicking, backspace, delete, typing etc No edit ability unless I'm missing something)

2,. Even leaving Credits as placeholder, and filling everything out, when I goto Episodes tab, the toggle switch stays red.. I cannot switch it to green. Going back to the Series tab, I cannot save it either, as a red warning box pops up saying 'Videos have not been defined for this content'  but again, I cannot seem to toggle Episodes to green to begin adding info.

**Reply by rrirower** [reply]
1. Above is a bug.  I'll code a fix for it.
2. Did you select the first Episode number first?  The fields are not enabled until you define the Episode number first.  This worked for me.

**Reply by DaveBarnitz** [reply]
2. Ok, that did it. Wasn't to intuitive to do it that way. I feel like I did anything except that. Any reason its using the Toggle switch there, instead of just an Error box at the end if anythings missing? 

Thanks tho, I could use redo-ing a Series.

**Reply by rrirower** [reply]
If you look at the Json Feed Manager you'll see that the toggle is the equivalent of the 'lock' button next to the content id.  Its sole purpose is to enable/disable that field.

**Reply by DaveBarnitz** [reply]
I noticed Episodes in a Series doesn't seem to save the episodes thumbnail url?   It validates and I see the preview, then I hit Save and when going back into it the Thumbnail field always seems to be blank? 

(I did download the new version too fwiw)

**Reply by rrirower** [reply]
I'll need to investigate this.

**Reply by rrirower** [reply]
I tried this with both a Series with Episodes only and, a Series with Seasons and Episodes.  Both saved the thumbnail url.  Are you making sure to click the save button after entering the attributes for the Series?

**Reply by DaveBarnitz** [reply]
Yeah, just tried it again. Everything else saves except that.  Does it validate internally at all?   I use Vimeo's thumbnail url, which doesn't end in a .jpeg or .png extension.  Is it looking for that?  (seems unlikely)

**Reply by rrirower** [reply]
Can you send me a sample of the url you are using so I can test it?

**Reply by DaveBarnitz** [reply]
I dm'd it to you on the Roku forum.

**Reply by rrirower** [reply]
Got it. I'll take a deeper dive into the code and try to debug why it doesn't seem to save.

**Reply by DaveBarnitz** [reply]
Thanks

**Reply by rrirower** [reply]
I'm still having trouble recreating this problem.  Can you please view the screen capture @ [DropBox](https://www.dropbox.com/s/e5z055xq1nxg48r/ScreenCaptureProject38.mp4?dl=0) and see if you are taking different steps?  This test was creating a Series with Episodes only. Are you creating Seasons?

**Reply by DaveBarnitz** [reply]
Just tried it again. Still isn't saving. I'll record a screenshare this weekend and upload it.

**Reply by DaveBarnitz** [reply]
No rush but here's a screen grab. You'll see I enter it into 1, 2, 3 and 5.  Then cycle back and they all seem gone? I am hitting Save and closing out after each one.  [https://drive.google.com/file/d/1NnX6wZIJxZiwEZ32wbJH-OhAI9sZpSD3/view?usp=share_link](url)

**Reply by rrirower** [reply]
I think I found the bug.  I should have a new version this weekend.  BTW...you don't need to toggle the switch on the right of the content ID.  It's sole purpose is to enable/disable the ID field.  IDs are supposed to be unique, so, they are locked after the object is created.

**Reply by DaveBarnitz** [reply]
Ok thanks. Yeah I wasn't sure if that was like an 'Edit' button. I'm not sure if I always did that, but when the glitch was occurring I was trying to gauge on best practice steps, and thought maybe thats what I was doing wrong. 

Nice catch finding whatever bug it was. For me, I would've just exported the feed and manually pasted the links into the json file directly (path of least resistance).  Wasn't trying to come off as a complainer, just felt there was something which to me, didn't seem to be working right, wanted to point it out.

**Reply by rrirower** [reply]
No worries.  It's important to get the software in front of as many eyes as possible.  That's the only real way to shake out the majority of bugs.

**Reply by DaveBarnitz** [reply]
Latest version this is fixed.  Will add some short forms & feature films and let you know if I encounter anything. I'd added a few previously & didn't seem to notice issues with that.

**Reply by rrirower** [reply]
If you get a chance, would you mind trying the Search feed generation process?  I'm interested in knowing if the resulting search feed passes the Roku validation tool.

**Reply by DaveBarnitz** [reply]
Sure thing. I'll be away this weekend tho but will pick it up next week

**Reply by DaveBarnitz** [reply]
FYI it generates, but to submit the  Roku validation tool I'd need to code out the entire channel. Wasn't really my intent at the moment, although I should do it at some point. I have a smaller channel I could try it out with, will still take some time as its 30 or so videos.

**Comment by axelctv** [comment]
just saw the message on the Roku dev forum - i would give it a try :)

**Comment by axelctv** [comment]
oh ....  please ad @gmail.com  after my id

**Reply by rrirower** [reply]
Please see your email for instructions.  Thanks.

**Reply by axelctv** [reply]
thanks
